Console/ReencryptColumn.php

```diff
@@ -0,0 +1,163 @@
+                null,
+                'The column that you want to re-encrypt',
+                ''
+            )
+        ];
+
+        $this->setName('gene:encryption-key-manager:reencrypt-column');
+        $this->setDescription('Re-encrypt a columns data with the latest key');
+        $this->setDefinition($options);
+
+        parent::configure();
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        if (!$input->getOption(self::INPUT_KEY_FORCE)) {
+            $output->writeln('<info>Run with --force to make these changes, this will run in dry-run mode by default</info>');
+        }
+
+        try {
+            $keys = preg_split('/\s+/s', trim((string)$this->deploymentConfig->get('crypt/key')));
+            $latestKeyNumber = count($keys) - 1;
+            $output->writeln("The latest encryption key is number $latestKeyNumber, looking for old entries");
+
+            $table = $input->getArgument(self::INPUT_KEY_TABLE);
+            if (!strlen($table)) {
+                throw new \Exception('Provide a table name');
+            }
+            if (in_array($table, ['core_config_data', 'tfa_user_config'])) {
+                throw new \Exception('You cannot use this command for this table');
+            }
+            $identifier = $input->getArgument(self::INPUT_KEY_IDENTIFIER);
+            if (!strlen($identifier)) {
+                throw new \Exception('Provide an identifier');
+            }
+            $column = $input->getArgument(self::INPUT_KEY_COLUMN);
+            if (!strlen($column)) {
+                throw new \Exception('Provide an column');
+            }
+            $output->writeln("Looking for '$column' in '$table', identified by '$identifier'");
+
+            /**
+             * @see \Magento\Framework\Model\ResourceModel\Db\AbstractDb::_getLoadSelect()
+             */
+            $tableName = $this->resourceConnection->getTableName($table);
+            $connection = $this->resourceConnection->getConnection();
+            $field = $connection->quoteIdentifier(sprintf('%s.%s', $tableName, $column));
+
+            $select = $connection->select()
+                ->from($tableName, [$identifier, "$column"])
+                ->where("($field LIKE '_:_:____%' OR $field LIKE '__:_:____%')")
+                ->where("$field NOT LIKE ?", "$latestKeyNumber:_:__%");
+
+            $result = $connection->fetchAll($select);
+            if (empty($result)) {
+                $output->writeln('No old entries found');
+                return Cli::RETURN_SUCCESS;
+            }
+            $connection->beginTransaction();
+            foreach ($result as $row) {
+                $output->writeln(str_pad('', 120, '#'));
+                $output->writeln("$identifier: {$row[$identifier]}");
+                $value = $row[$column];
```

Comment:
Make sense to add something like that for NULL values?
```suggestion
                $value = $row[$column];
                if ($value === null) {
                    $output->writeln("The row #{$row[$identifier]} is NULL, skipped");
                    continue;
                }
```